src/utils/Router.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const express = require('express');
+
+class Router {
+  constructor(controller, ...middlewaresAndRoutes) {
+    const routes = middlewaresAndRoutes.at(-1);
+    const middlewares = middlewaresAndRoutes.slice(0, -1);
+
+    this.router = express.Router();
+    middlewares.forEach(middleware => this.router.use(middleware));
+
+    routes.forEach(route => {
+      const { method, path, handler } = route;
+
+      this.router[method](path, handler.bind(controller));
+    });
+  }
+}
```

Comment:
this looks buggy to be honest
consider to do this:
```suggestion
class Router {
  constructor({ controller, middlewares, routes }) {
    this.router = express.Router();
    middlewares.forEach(middleware => this.router.use(middleware));

    routes.forEach(route => {
      const { method, path, handler } = route;

      this.router[method](path, handler.bind(controller));
    });
  }
}
```

and then usage:
```js
const router = new Router({
  controller: {},
  middlewares: [],
  routes: {}
})
```